nitro-protocol/contracts/CurrencySwapApp.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/IForceMoveApp.sol';
+import './libraries/NitroUtils.sol';
+import './interfaces/INitroTypes.sol';
+import {ExitFormat as Outcome} from '@statechannels/exit-format/contracts/ExitFormat.sol';
+
+/**
+ * @dev The CurrencySwapApp contract complies with the ForceMoveApp interface and allows Alice and Bob (participants[0] to participants[n+1], where n is the number of intermediaries) to trade two assets atomically.
+ */
+contract CurrencySwapApp is IForceMoveApp {
+    struct Order {
+        address asset;
+        uint256 amount;
+    }
+
+    struct Voucher {
+        Order takerOrder;
+        Order makerOrder;
+        // TODO: I assume both side need to confirm the amount
+        INitroTypes.Signature takerSignature;
+        INitroTypes.Signature makerSignature;
+        // TODO: to differentiate between Vouchers
```

Comment:
We need a new convention, but we can try to use the Initiator / Receiver
We are now in single currency, but bi-directional,

which mean two Signatures.